src/tests/ftest/util/osa_utils.py

```diff
@@ -36,6 +37,7 @@ class OSAUtils(IorTestBase):
     def setUp(self):
         """Set up for test case."""
         super(OSAUtils, self).setUp()
+        self.pool_cont_dict = {}
         self.container = None
```

Comment:
Minor: If you want to avoid dealing with too many variables you can initialize self.pool and self.container both as list. And every time you create a pool a container can be appended to that list and later accessed accordingly.  This will also take care of itself in teardown.

Reply:
I like this dictionary method for now... When I move self.pool to list and I will definitely take a look at it...